cfme/infrastructure/deployment_roles.py

```diff
@@ -1,67 +1,266 @@
 """ A model of an Infrastructure Deployment roles in CFME"""
 
 from functools import partial
-from navmazing import NavigateToAttribute, NavigateToSibling
-from selenium.common.exceptions import NoSuchElementException
 
-from cfme.exceptions import ListAccordionLinkNotFound
-from cfme.fixtures import pytest_selenium as sel
+from navmazing import NavigateToAttribute, NavigateToSibling
+from widgetastic.exceptions import NoSuchElementException
+from widgetastic.utils import Version, VersionPick
+from widgetastic.widget import ParametrizedView, View
+from widgetastic_manageiq import (
+    Accordion,
+    BaseEntitiesView,
+    BaseListEntity,
+    BaseQuadIconEntity,
+    BaseTileIconEntity,
+    BootstrapSelect,
+    BootstrapTreeview,
+    BreadCrumb,
+    ItemsToolBarViewSelector,
+    JSBaseEntity,
+    NonJSBaseEntity,
+    PaginationPane,
+    SummaryTable,
+    Table,
+    Text
+)
+from widgetastic_patternfly import (
+    BootstrapNav,
+    Button,
+    Dropdown,
+    FlashMessages
+)
+
+from cfme.base.ui import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound, RoleNotFound
 from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+from utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
 from utils.appliance import Navigatable
-from utils.appliance.implementations.ui import (CFMENavigateStep, navigate_to,
-                                                navigator)
-from cfme.web_ui import (listaccordion as list_acc, match_location, Quadicon,
-                         Region, toolbar as tb)
-from utils import version
-from utils.pretty import Pretty
 
 
-details_page = Region(infoblock_type='detail')
+class DeploymentRoleToolbar(View):
+    """The toolbar on the Deployment Role page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    download = Dropdown('Download')
+    view_selector = View.nested(ItemsToolBarViewSelector)
+
+
+class DeploymentRoleDetailsToolbar(View):
+    """The toolbar on the Deployment Role details page"""
+    configuration = Dropdown('Configuration')
+    policy = Dropdown('Policy')
+    monitoring = Dropdown('Monitoring')
+    download = Button('Download summary in PDF format')
```

Comment:
Hmm, shouldn't it be `Button(title='Download ....')` since the download buttons have no visible text on them?

Reply:
yes, @mfalesni . It has only pdf Icon not text on a button.